src/server.py

```diff
@@ -21,5 +24,12 @@
 
 logging.info("Home folder: {}".format(KOLIBRI_HOME))
 logging.info("Timezone: {}".format(os.environ.get("TZ", "(default)")))
+
+args = {}
+_, port, _, status = _read_pid_file(PID_FILE)
+# Possibly a killed process so we try to use the same port
+if status == STATUS_RUNNING and port:
+    args["port"] = port
```

Comment:
This bugged me a little because I took the time to make Kolibri _not_ do this a while ago (it was reusing random port numbers for a different reason). But it definitely makes sense why you're doing it here since passing that new port number to the main process and getting it to do the right thing is pretty complex. Good fix :)

I created a Phabricator task about handling service restarts more directly. Low priority, not attached to anything - just to keep track in case we need it: https://phabricator.endlessm.com/T33761.